src/js/modal/openModal.js

```diff
@@ -22,39 +33,65 @@ const openModal = e => {
       document.body.appendChild(backdrop);
 
       // blokowanie scrollowania strony
-      const currentScrollY = window.scrollY;
-      document.body.style.top = `-${currentScrollY}px`;
-      document.body.style.position = 'fixed';
+      handleBodyScrolling(true, currentScrollY);
 
       const closeModalButton = document.querySelector('#modal__close');
       const watchedButton = document.getElementById('watched');
       const queueButton = document.getElementById('queue');
 
-      watchedButton.addEventListener('click', () =>
-        addToLocalStorage(movieId, 'watched')
-      );
-      queueButton.addEventListener('click', () =>
-        addToLocalStorage(movieId, 'queue')
-      );
+      toLocalButton(watchedButton, movieId, 'watched', isWatched);
+      toLocalButton(queueButton, movieId, 'queued', isQueued);
+
       closeModalButton.addEventListener('click', () => {
         // usuwanie nasłuchiwaczy zdarzeń
-        watchedButton.removeEventListener('click', () =>
-          addToLocalStorage(movieId, 'watched')
-        );
-        queueButton.removeEventListener(
-          'click',
-          addToLocalStorage(movieId, 'queue')
-        );
+        watchedButton.removeEventListener('click', () => null);
+        queueButton.removeEventListener('click', null);
         // odblokowanie scrollowania strony
-        document.body.style.position = '';
-        document.body.style.top = '';
-        window.scrollTo(0, currentScrollY);
+        handleBodyScrolling(false, currentScrollY);
         closeModal();
       });
     })
     .catch(error => {
-      console.error(error);
+      throw new Error(error);
```

Comment:
jeśli to jest catch który nie obsługuje` !response.ok` to wystarczy chyba zwykły console.error - sprawdź sobie jeśli masz czas